test/functional/job/job_spec.lua

```diff
@@ -104,6 +104,17 @@ describe('jobs', function()
     nvim('eval', 'jobsend(j, "abcdef")')
     nvim('eval', 'jobstop(j)')
     eq({'notification', 'j', {{jobid, 'stdout', {'abcdef'}}}}, next_message())
-    eq({'notification', 'j', {{jobid, 'exit'}}}, next_message())
+    eq({'notification', 'j', {{jobid, 'exit', 0}}}, next_message())
+  end)
+
+  it('will return the proper exit status for the job', function()
+    nvim('command', notify_job())
+    nvim('command', "let j = jobstart('xxx', 'cat', ['/dev/null/fake'])")
+    local jobid = nvim('eval', 'j')
+    next_message()
+    eq({'notification', 'j', {{jobid, 'exit', 1}}}, next_message())
+    nvim('command', "let j = jobstart('xxx', 'bash', ['-c', 'exit 100'])")
+    local jobid = nvim('eval', 'j')
+    eq({'notification', 'j', {{jobid, 'exit', 100}}}, next_message())
```

Comment:
good point. It could be tested with unit tests that call the C functions directly, but seems like the related functions are static, so won't be easy.